state/txindex/indexer_service.go

```diff
@@ -119,9 +137,82 @@ func (is *IndexerService) OnStart() error {
 	return nil
 }
 
+func (is *IndexerService) Prune(retainHeight int64) {
+	lastRetainHeight, err := is.getLastIndexerRetainHeight()
+	if err != nil {
+		panic(err)
+	}
+	if retainHeight <= lastRetainHeight {
+		return
+	}
+	is.txIdxr.Prune(lastRetainHeight, retainHeight)
+	is.blockIdxr.Prune(lastRetainHeight, retainHeight)
+	is.setLastIndexerRetainHeight(retainHeight)
+}
+
+func (is *IndexerService) SaveIndexerRetainHeight(height int64) error {
+	return is.indexerStore.SetSync(IndexerRetainHeightKey, int64ToBytes(height))
+}
+
+func (is *IndexerService) GetIndexerRetainHeight() (int64, error) {
+	buf, err := is.getValue(IndexerRetainHeightKey)
+	if err != nil {
+		return 0, err
+	}
+	height := int64FromBytes(buf)
+
+	if height < 0 {
+		return 0, ErrInvalidHeightValue
+	}
+
+	return height, nil
+}
+
+func (is *IndexerService) getLastIndexerRetainHeight() (int64, error) {
+	bz, err := is.getValue(LastIndexerRetainHeightKey)
```

Comment:
I think that, instead of making this a generic 'GetValue` , you should have a `getLastIndexerRetainHeight` where you explicitly query only for that key. I know these functions are all private so users cannot misuse them, but as this is the pattern followed elsewhere it might make more sense.